docs/changelog.md

```diff
@@ -1,3 +1,139 @@
+
+- MOD files are compiled as C++ instead of C (see [migration guide](https://nrn.readthedocs.io/en/latest/guide/porting_mechanisms_to_cpp.html#porting-mechanisms-to-cpp) for VERBATIM blocks)  
+- API changes in the functions related to random numbers, see #2618  
+- Restored usage of TABLE statement in hh.mod, results are now same as 8.0.2 (see #1764)  
+- NMODL: Disallow use of ion variable in the CONSTANT block (#1955)  
+- NMODL: Disallow declaring variables and functions with the same name (#1992)  
+- Usage of Eigen library could introduce floating point differences, but they are very small, validated and can be ignored  
+
+### Removal / Deprecation
+
+- Removed HOC function parent_node.(Issue 3571) (#3576)
+- Removed usage of legacy OS X carbon libraries (#1869)  
+- Removed legacy LINDA code and Java bindings (#1919, #1937)  
+- Removed unused NMODL methods: adams, heun, clsoda, seidel, simplex, gear (#2032)  
+- Removed support for Python 3.7 and 3.8 (#2194, #3108)  
+- Removed support for legacy units and dynamic units selection (#2539)  
+- Removed unused NEMO simulator compatibility (#3035)  
+- Removed checkpoint feature (`h.checkpoint()`), superseded by SaveState (#3092)  
+- Removed legacy Random.MLCG and Random.ACG random number generators (#3189, #3190)  
+- Removed support for mod2c transpiler for CoreNEURON (#2247)  
+- Removed unused NMODL constructs: PUTQ, GETQ, RESET, MATCH, TERMINAL, SECTION,  
+  IFERROR, MODEL_LEVEL, PLOT, SENS, etc. (#1956, #1974, #1975, #2001, #2004, #2005)  
+- Removed `__MWERKS__` (CodeWarrior compiler) related code (#2655)  
+- Removed obsolete uxnrnbbs code (#2203)  
+- Removed old tqueue implementations (#2225, #2740)  
+- Removed unused RxD code_matrix_to_rxd_sparse (#3547)
+- Py_NoSiteFlag is deprecated for site_import (#3286)
+
+### Bug Fixes
+
+- Fix segfault from unref hoc_obj, see #1857 (#1917)  
+- Fix bug in rxd reactions involving rxd parameters (#1933)  
+- Fix CoreNEURON bug when checkpointing VecPlay instances (#2148)  
+- Fix nocmodl bug with unused STATE + COMPARTMENT variable in KINETIC block (#2210)  
+- Fix for dynamic ECS diffusion characteristics (#2229)  
+- Various fixes for compatibility with newer NVHPC compiler releases (#2239, #2591)  
+- Fix Windows and Mac crash on multiline HOC statements input from terminal (#2258)  
+- Fix for Vector.max_ind (#2251)  
+- Fix for detecting ION channel type in LoadBalance (#2310)  
+- Fix for RangeVarPlot adding extra point at section ends when using plotly (#2410)  
+- Fixed mod file array variable assignment (seg.mech.array[i] = x) (#2504)  
+- Various fixes in RxD features (#2593, #2588)  
+- Fix CoreNEURON bug causing in-memory model copy with file mode (#2767)  
+- Fix CoreNEURON bug when using array variables in MOD files (#2779)  
+- Fix lexer for ONTOLOGY parsing (#3091)  
+- Fix issue while using Anaconda Python on MacOS (#3088)
+- Fix ParallelContext bbs memory leaks (#3563)
+- Fix RxD indexing bug? (#3337)
```

Comment:
Probably should leave out the question mark, unless we are sure that it could still appear, in which case, rephrasing the statement is still better than using a question mark.

Reply:
Done